internal/pkg/gateway/commit/notifier_test.go

```diff
@@ -0,0 +1,285 @@
+/*
+Copyright 2021 IBM All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package commit_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/hyperledger/fabric-protos-go/peer"
+	"github.com/hyperledger/fabric/core/ledger"
+	"github.com/hyperledger/fabric/internal/pkg/gateway/commit"
+	"github.com/hyperledger/fabric/internal/pkg/gateway/commit/mock"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+)
+
+//go:generate counterfeiter -o mock/notificationsupplier.go --fake-name NotificationSupplier . NotificationSupplier
+
+func TestNotifier(t *testing.T) {
+	newTestNotifier := func(commitSend <-chan *ledger.CommitNotification) *commit.Notifier {
+		notificationSupplier := &mock.NotificationSupplier{}
+		notificationSupplier.CommitNotificationsReturnsOnCall(0, commitSend, nil)
+		notificationSupplier.CommitNotificationsReturns(nil, errors.New("unexpected call of CommitNotificationChannel"))
+
+		return commit.NewNotifier(notificationSupplier)
+	}
+
+	t.Run("NewNotifier with nil ledger returns nil", func(t *testing.T) {
+		result := commit.NewNotifier(nil)
+		require.Nil(t, result)
+	})
+
+	t.Run("Notify", func(t *testing.T) {
+		t.Run("returns error from notification supplier", func(t *testing.T) {
+			notificationSupplier := &mock.NotificationSupplier{}
+			notificationSupplier.CommitNotificationsReturns(nil, errors.New("MY_ERROR"))
+			notifier := commit.NewNotifier(notificationSupplier)
+			defer notifier.Close()
+
+			_, err := notifier.Notify(context.Background().Done(), "CHANNEL_NAME", "TX_ID")
```

Comment:
Same question here about the use of context at this call and all of the others in the test

Reply:
I wanted this code to be easy to use for any calling code that was using context, by having the same done channel type. This was my way of ensuring I didn't mess up. It doesn't need to use context so I have removed it 👍